context.go

```diff
@@ -119,8 +122,29 @@ func applyConnMetadata(ctx Context, conn gossh.ConnMetadata) {
 	ctx.SetValue(ContextKeyRemoteAddr, conn.RemoteAddr())
 }
 
+func (ctx *sshContext) parentContext() context.Context {
+	return *ctx.parentCtx.Load().(*context.Context)
+}
+
+func (ctx *sshContext) Deadline() (deadline time.Time, ok bool) {
+	return ctx.parentContext().Deadline()
+}
+
+func (ctx *sshContext) Done() <-chan struct{} {
+	return ctx.parentContext().Done()
+}
+
+func (ctx *sshContext) Err() error {
+	return ctx.parentContext().Err()
+}
+
+func (ctx *sshContext) Value(key interface{}) interface{} {
+	return ctx.parentContext().Value(key)
+}
+
 func (ctx *sshContext) SetValue(key, value interface{}) {
-	ctx.Context = context.WithValue(ctx.Context, key, value)
+	newCtx := context.WithValue(ctx.parentContext(), key, value)
+	ctx.parentCtx.Store(&newCtx)
```

Comment:
That's still not safe if A and B both calls `SetValue` at the same time only one value will be stored.

Reply:
Good point, although it is the old behavior, it's not ideal ... Will rewrite it.

Reply:
Done in d9f5b41d17848bfbfc072ec9c3814ec77f3e78a0